en/developer_guide/bbcode.rst

```diff
@@ -0,0 +1,173 @@
+   instance in the media case, you will see it in the editor.
+4. When the user saves the form, TinyMce replaces via javascript the html tags by their BBCode
+   representations.
+
+How to create a new BBCode and add it to the parser
+---------------------------------------------------
+Open Orchestra provides a simple ways of adding your own custom BBCode tags, depending on their
+nature. Two kind of tags exists:
+
+- Simple tags are just text replacements. For instance a ``[b]Some Text[/b]`` should be replaced by
+``<b>Some Text</b>``
+
+- Complex tags require more than a simple regexp conversion. For instance the tag
+``[media=format]MediaId[/media]`` requires to fetch the media in the database, to detect its type
+and then to to return the corresponding html tag.
+
+To add a new BBCode to the parser, you need to create its definition. There are two ways of adding
+a definition. To create a simple tag you can quickly set a configuration. But if you prefer, or if
+your tag requires a php analysis, you can use a tagged service.
+
+Create a simple BBCode tag via configuration
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+To create a simple BBCode definition, its possible to use a trivial configuration. The parameter
+``open_orchestra_bbcode.code_definitions`` contains an array of declared simple tags. Here is the
+declaration of the bold tag:
+
+.. code-block:: yaml
+
+    bold:
+        tag: b
+        html: <strong>{param}</strong>
+
+- The key ``bold`` is the name of the definition used internally by the parser.
+
+- The ``html`` key is the html pattern of the code. The string ``{param}`` will be replaced by the
+content of the BBcode tag.
+
+- The ``tag`` key is the label used in the BBCode tag. In this example, with the ``tag`` key set to
+``b``, the BBCode tag is ``[b]{param}[/b]``.
+
+Simple tags can also include some options. For instance the tag ``[h]`` which matches a ``<h*>``
+html tag requires an option to indicate the level of the header. For instance ``[h=1]Some text[/h]``
+will produce a ``<h1>`` html tag. You still can declare this in a configuration file like this:
+
+.. code-block:: yaml
+
+        title:
+            tag: h
+            html: <h{option}>{param}</h{option}>
+            parameters:
+                use_option: true
+
+To declare that a tag uses option, you must add the key ``parameters.use_option`` and set it to true.
+Note that when declaring a tag with option, that option becomes mandatory. So if the option is not
+mandatory, you should declare two versions of the tag: one with option and one without. That way
+the parser will use one or the other according to the presence of an option. The ``html`` key uses
+both {param} and {option} string to indicate where to place those informations.
+The resulting BBCode will be ``[h={option}]{param}[/h]``
+
+Create a BBCode tag via a service
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+BBCode definitions can also be declared by services. A compiler pass has been added to the
```

Comment:
I don't think that talking about `CompilerPass` here is usefull to the documentation.

You could say : 
`The service receiving the symfony tag ... are going to be added to the parser`